tests/bsim/bluetooth/audio/src/public_broadcast_sink_test.c

```diff
@@ -0,0 +1,423 @@
+
+	if (!bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO)) {
+		/* Save broadcast_id */
+		if (broadcast_id == INVALID_BROADCAST_ID) {
+			broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
+		}
+
+		/* Found Broadcast Audio and Public Broadcast Announcement Services */
+		if (pbs_found) {
+			return false;
+		}
+	}
+
+	if (!bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_PBA)) {
+		bt_pbp_parse_public_broadcast_announcement(data, &source_features, meta);
+		if (!(source_features & BT_PBP_ANNOUNCEMENT_FEATURE_HIGH_QUALITY)) {
+			/* This is a Standard Quality Public Broadcast Audio stream - do not sync */
+			printk("This is a Standard Quality Public Broadcast Audio stream\n");
+			pbs_found = false;
+			return true;
+		}
+		printk("Found Suitable Public Broadcast Announcement\n");
+		pbs_found = true;
+
+		/* Continue parsing if Broadcast Audio Announcement Service was not found */
+		if (broadcast_id == INVALID_BROADCAST_ID) {
+			return true;
+		}
+		return false;
+	}
+
+	/* Continue parsing */
+	return true;
+}
+
+static void broadcast_scan_recv(const struct bt_le_scan_recv_info *info,
+				struct net_buf_simple *ad)
+{
+	pbs_found = false;
+
+	/* We are only interested in non-connectable periodic advertisers */
+	if ((info->adv_props & BT_GAP_ADV_PROP_CONNECTABLE) ||
+	     info->interval == 0) {
+		return;
+	}
+
+
+	bt_data_parse(ad, scan_check_and_sync_broadcast, (void *)&broadcast_id);
+
+	if ((broadcast_id != INVALID_BROADCAST_ID) && pbs_found) {
+		sync_broadcast_pa(info);
+	}
+}
+
+static void test_main(void)
+{
+	int count = 0;
+	int err;
+
+	init();
+
+	while (count < 2) {
+		err = reset();
+		if (err != 0) {
+			printk("Resetting failed: %d\n", err);
+			break;
+		}
+
+		/* Register callbacks */
+		bt_le_scan_cb_register(&broadcast_scan_cb);
+
+		/* Start scanning */
+		err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, NULL);
+		if (err) {
+			printk("Scan start failed (err %d)\n", err);
+			break;
+		}
+
+		/* Wait for PA sync */
+		err = k_sem_take(&sem_pa_synced, SEM_TIMEOUT);
+		if (err != 0) {
+			printk("sem_pa_synced timed out\n");
+			break;
+		}
+
+		/* Wait for BASE decode */
+		err = k_sem_take(&sem_base_received, SEM_TIMEOUT);
+		if (err != 0) {
+			printk("sem_base_received timed out\n");
+			break;
+		}
+
+		/* Create broadcast sink */
+		err = bt_bap_broadcast_sink_create(bcast_pa_sync, broadcast_id, &broadcast_sink);
+		if (err != 0) {
+			printk("Sink not created!\n");
+			break;
+		}
+
+		k_sem_take(&sem_syncable, SEM_TIMEOUT);
+		if (err != 0) {
+			printk("sem_syncable timed out\n");
+			break;
+		}
+
+		/* Sync to broadcast source */
+		err = bt_bap_broadcast_sink_sync(broadcast_sink, bis_index_bitfield,
+						streams_p, NULL);
+		if (err != 0) {
+			printk("Unable to sync to broadcast source: %d\n", err);
+			break;
+		}
+
+		k_sem_take(&sem_pa_sync_lost, SEM_TIMEOUT);
+
+		count++;
+	}
+
+	if (count == 1) {
+		/* Pass if we synced only with the high quality broadcast */
+		PASS("Public Broadcast sink passed\n");
+	}
```

Comment:
```suggestion
	if (count == 1) {
		/* Pass if we synced only with the high quality broadcast */
		PASS("Public Broadcast sink passed\n");
	} else {
		FAIL("Public Broadcast sink failed\n");
	}
```

Alternative, replace `break` with `FAIL` in the above loop. 

This also applies for the broadcast source test